File: web/webTrung/java_spring_-thymeleaf_food/api/src/main/java/doan/_java_food/models/Comment.java
package doan._java_food.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Entity
@Table(name = "comments")
@Getter
@Setter
public class Comment {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name =  "status", columnDefinition = "VARCHAR(255) DEFAULT 'APPROVED'")
    private String status;

    @Column(name =  "email")
    private String email;

    @Column(name =  "name")
    private String name;

    @Column(name =  "vote_number")
    private Integer voteNumber;

    @Column(name =  "phone")
    private String phone;

    @Column(columnDefinition = "TEXT", name =  "content")
    private String content;

    @ManyToOne
    @JoinColumn(name = "user_id",referencedColumnName = "id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "restaurant_id",referencedColumnName = "id")
    private Restaurant restaurant;


    @Column(name = "created_at")
    private Date createdAt;

    @Column(name = "updated_at")
    private Date updatedAt;
}

File: app/pasgo_restaurants/lib/domain/entities/comment/comment.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pasgo_restaurants_ecommerce/data/models/restaurant/uu_model.dart';
import 'package:pasgo_restaurants_ecommerce/domain/entities/restaurant/restaurant.dart';

part 'comment.g.dart';

@JsonSerializable()
class Comment {
  final int id;
  final String status;
  final String email;
  final String name;
  // final int? voteNumber;
  final String? phone;
   String content;
  final String? createdAt;
  final String? updatedAt;
  final UUserModel user;
  final Restaurant restaurant;

   Comment({
    required this.id,
    required this.name,
    required this.status,
    required this.email,
    // this.voteNumber,
    this.phone,
    required this.content,
    this.createdAt,
    this.updatedAt,
    required this.user,
    required this.restaurant,
  });

  factory Comment.fromJson(Map<String, dynamic> json) =>
      _$CommentFromJson(json);

  Map<String, dynamic> toJson() => _$CommentToJson(this);
}
